hazelcast/src/main/java/com/hazelcast/nio/serialization/genericrecord/GenericRecordBuilder.java

```diff
@@ -119,14 +121,16 @@ static GenericRecordBuilder compact(String typeName) {
      *                  characters for {@link Portable} GenericRecords.
      * @param value     to set to GenericRecord
      * @return itself for chaining
-     * @throws HazelcastSerializationException if the field name does not exist
+     * @throws HazelcastSerializationException 1. if the field name does not exist
      *                                         in the schema/class definition or
      *                                         the type of the field does not
      *                                         match the one in the schema/class
      *                                         definition or the same field is
      *                                         trying to be set without using
      *                                         {@link
      *                                         GenericRecord#newBuilderWithClone()}.
+     *                                         2. if the field is set after the
```

Comment:
I think it makes sense but I remember at first I designed it as another exception than serialization exception but later on I changed it again for some reason. I don't remember what the reason was but I can try again. @mdumandag Do you agree with @ihsandemir here?

Reply:
I think `UnsupportedOperationException ` makes a little bit more sense. We were using `HazelcastSerializationException` for similar errors in Portable, (see `ClassDefinitionBuilder`), so this will be inconsistent with it, but it is a minor thing.

Reply:
Yeah I think that was the reason. Ok let me change it.